File: Find Bottom Left Tree Value.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func findBottomLeftValue(root *TreeNode) int {
    max_depth := 0
    val := root.Val
    var exploreNode func(node *TreeNode, depth int)

    exploreNode = func(node *TreeNode, depth int) {
        if node == nil {
            return
        }

        if depth > max_depth {
            max_depth = depth
            val = node.Val
        }

        exploreNode(node.Left, depth+1)
        exploreNode(node.Right, depth+1)
    }

    exploreNode(root, 0)
    return val
}

File: Find Bottom Left Tree Value.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        max_depth, val = 0, root.val

        def exploreNode(node, depth):
            nonlocal max_depth, val
            if node is None:
                return

            if depth > max_depth:
                max_depth = depth
                val = node.val

            exploreNode(node.left, depth + 1)
            exploreNode(node.right, depth + 1)

        exploreNode(root, 0)
        return val
